src/components/WidgetHolder/WidgetHolder.tsx

```diff
@@ -22,7 +21,15 @@ const WidgetHolder = () => {
 	const getWidgetDOM = (widgetData: WidgetData, index: number) => {
 		switch (widgetData.type) {
 			case WidgetType.BookmarkWidget:
-				return <BookmarkWidget key={index} {...widgetData} />;
+				return (
+					<BookmarkWidget
+						key={index}
+						index={index}
+						widgetsData={widgetsData}
```

Comment:
```suggestion
						{...widgetsData}
```

Reply:
@yashchaudhari008 We will need to pass this `widgetsData` as it is without destructuring. Because it is an array and used to filter out the widget to be deleted or edited.